dunwoody_disaster/CharacterFactory.py

```diff
@@ -252,28 +254,33 @@ def createTestChar() -> Character:
         character = CharacterFactory.createCharacter("Test-Char", "blank")
         return character
 
+    @staticmethod
+    def saveCharacter(character: Character) -> None:
+        """
+        Saves a character to a json file
+        :param character: The character to save
+        """
+        with open(f"{character.name}.json", "w") as f:
+            json.dump(character.serialize(), f)
+
+    @staticmethod
+    def loadCharacter(name: str) -> Character:
+        """
+        Loads a character from a json file
+        :param name: The name of the character to load
+        :return: The loaded character object
+        """
+        if not os.path.exists(f"{name}.json"):
+            raise FileNotFoundError(f"Character file {name}.json not found")
+        with open(f"{name}.json", "r") as f:
+            data = json.loads(f.read())
+            character = CharacterFactory.createFromJson(data)
+            return character
+
     @staticmethod
     def createFromJson(json: dict) -> Character:
         char = Character()
-        char.name = json["name"]
-        char.image_path = DD.ASSETS[json["asset"]]
-        char.level = json["level"]
-        char.classType = json["class"]
-        char.strength = json["strength"]
-        char.intelligence = json["intelligence"]
-        char.maxHealth = json["health"]
-        char.maxMagic = json["magic"]
-        char.maxStamina = json["stamina"]
-        char.defense = json["defense"]
-        char.magicDefense = json["magicDefense"]
-
-        char.set_health(json["health"])
-        char.set_magic(json["magic"])
-        char.set_stamina(json["stamina"])
-
-        for item in json["inventory"]["weapons"]:
-            char.add_item(Item.Weapon.from_json(item))
-        for item in json["inventory"]["defenses"]:
-            char.add_item(Item.Armor.from_json(item))
+        for key, value in json.items():
+            setattr(char, key, value)
```

Comment:
This is so handy! Is there a way to do it in the `.serialize` method?

Reply:
lmao i just asked chatGPT
`    def serialize(self):
        serialized_data = {}
        # Iterate through object attributes
        for key, value in self.__dict__.items():
            # Exclude special attributes and methods (starting with "__")
            if not key.startswith("__") and not callable(value):
                serialized_data[key] = value
        return serialized_data`

Reply:
I can't even understand that lol.